tests/SConscript

```diff
@@ -5,25 +5,49 @@ import os
 testenv = env.Clone()
 
 testenv.Replace(
-    CPPPATH= ['../src', '../utils/include',
+    CPPPATH= ['../src', '../include', '../utils/include', '../src/vcl/',
               '/usr/include/jsoncpp/',
               os.path.join(env['INTEL_PATH'], 'pmgd/include'),
               os.path.join(env['INTEL_PATH'], 'pmgd/util'),
-              os.path.join(env['INTEL_PATH'], 'vcl/include'),
-              os.path.join(env['INTEL_PATH'], 'vcl/src'),
              ])
 
 testenv.Append(
-        LIBS = ['gtest'],
+        LIBS = ['gtest',
+                'tiledb',
+                'opencv_core',
+                'opencv_imgproc',
+                'opencv_imgcodecs',
+                'opencv_videoio',
+                'opencv_highgui',
+                'gomp',
+                'faiss',
+                'gtest', 'pthread',
+              ],
+
         LIBPATH = ['../utils/']
         )
 
 testenv.MergeFlags(GetOption('cflags'))
 
-test_sources = ['main.cc', 'pmgd_queries.cc', 'json_queries.cc']
+test_sources = ['main.cc',
+                'unit_tests/pmgd_queries.cc',
+                'unit_tests/json_queries.cc',
+
+                'unit_tests/helpers.cc',
+                'unit_tests/TDBImage_test.cc',
+                'unit_tests/ImageData_test.cc',
+                'unit_tests/Image_test.cc',
+                'unit_tests/Video_test.cc',
+                'unit_tests/VideoData_test.cc',
+                'unit_tests/DescriptorSetAdd_test.cc',
+                'unit_tests/DescriptorSetClassify_test.cc',
+                'unit_tests/DescriptorSetTrain_test.cc',
+                'unit_tests/DescriptorSetReadFS_test.cc',
+                'unit_tests/DescriptorSetStore_test.cc',
+              ]
 
 query_tests = testenv.Program(
-                'query_tests',
+                'unit_test',
```

Comment:
I think this commit is very mixed up. If you want to split the tests into unit tests and general tests, then that should go before merging vcl. Also, json_queries.cc is not a unit test. It tests multiple components at the same time. pmgd_queries.cc is a unit test.

Reply:
Also, I think the organization for tests now needs to segregate tests/unit_tests and tests/server. Then inside server, you can have python and cpp. That will capture the situation better I feel.

Reply:
we will keep the test merge in the same commit, we will separate the tests folder as we discussed.